model/host/host.go

```diff
@@ -2268,23 +2282,31 @@ func FindTerminatedHostsRunningTasks() ([]Host, error) {
 }
 
 // CountContainersOnParents counts how many containers are children of the given group of hosts
-func (hosts HostGroup) CountContainersOnParents() (int, error) {
+func (hosts HostGroup) CountContainersOnParents(ctx context.Context) (int, error) {
 	ids := hosts.GetHostIds()
-	query := db.Query(mgobson.M{
-		StatusKey:   mgobson.M{"$in": evergreen.UpHostStatus},
-		ParentIDKey: mgobson.M{"$in": ids},
-	})
-	return Count(query)
+	if len(ids) == 0 {
```

Comment:
This was just to fix the query because it's not valid to pass an array to `$in` [here](https://github.com/evergreen-ci/evergreen/blob/a8dcd454b1ec46dead33677c2cfdabce2269c55e/model/host/host.go#L2293) while if the group was empty we are passing `nil`/`null` ([this is a test that hit it](https://evergreen.mongodb.com/test_log/evergreen_ubuntu2204_test_model_host_patch_c74d0fb99bd582cd832ead7dac8a8777ed4cdc14_64a32c0be3c331304c674967_23_07_03_20_14_04/0?test_name=output.model-host.test&group_id=#L555)).

This was valid before for a hacky reason: the query used to use [an mgobson map](https://github.com/evergreen-ci/evergreen/blob/e1789d96a1729ae60ad9dd48d6401bb9a2f16cd5/db/mgo/bson/bson.go#L113), which [uses mgo's Marshal](https://github.com/evergreen-ci/evergreen/blob/e1789d96a1729ae60ad9dd48d6401bb9a2f16cd5/db/mgo/bson/extensions.go#L5) which apparently filled in the nil array as an empty one. Now it's using the mongo driver's registry which does not. (I didn't dig through the code to verify this, but it seems to be the case)